csi.proto

```diff
@@ -492,6 +468,15 @@ message ControllerServiceCapability {
       PUBLISH_UNPUBLISH_VOLUME = 2;
       LIST_VOLUMES = 3;
       GET_CAPACITY = 4;
+      CREATE_DELETE_SNAPSHOT = 5;
+      // For storage systems that need to upload the snapshot after
+      // it is cut, LIST_SNAPSHOTS is a required RPC because CO needs
+      // to call it to find out whether the upload is complete.
+      LIST_SNAPSHOTS = 6;
+      // Currently the only way to consume a snapshot is to create
+      // a volume from it. In the future, a revert snapshot function
+      // may be added as another way to consume the snapshot.
+      CREATE_VOLUME_FROM_SNAPSHOT = 7;
```

Comment:
Do we have a solid case for drivers that support snapshots that wouldn't support this option?

Reply:
@cpuguy83 Do you mean the "revert" case?  I do think there are a number of backends that have trouble with this; Older versions of Cinder (and a number of drivers in Cinder still) have trouble with this one.  

I also think the common provider model is create-from-xxxx, during the Snapshot discussions way back we had a lot of discussion around this and I think we had some pretty solid concensus on the revert being a unique call.  

Of course if that's not what you mean here.. then.. well... never mind :)

Reply:
@cpuguy83 I don't have a solid case for drivers that support snapshots that wouldn't support "CREATE_VOLUME_FROM_SNAPSHOT".  Should we require drivers supporting CREATE_DELETE_SNAPSHOT to support create volume from snapshot?